### PR TITLE
Generate the site through antwar

### PR DESCRIPTION
**Pull request by bebraw**
This PR generates the site through [antwar](https://github.com/antwarjs/antwar), a site generator of mine (webpack/react). It's quite full-featured and I'll be adding functionality to it as we need it.

A couple of notes:

* The styling looks rough (headers etc. are off) but I expect @gregvenech will take care of those so I didn't put too much thought into that. The files can move around etc. based on the need. It's just webpack underneath after all.
* Most of the focus was put on content. I restructured the site. From contributor's point of view it's just markdown (see `content/`) and I tried to keep it as neat as possible. You can see this in the proposed sectioning.
* Section index and pages (root/section) can be styled and modified separately through `layouts/`. I made it so that section pages have Disqus in them and also contain previous/next links for easier navigation. Custom functionality can be added there as needed.
* Each page contains Gitter sidecar pointing to the official chat. Easy to disable if you want, though. This is managed through `layouts/Body.jsx`. You can also control the GitHub "fork" banner, and site navigation there.
* There's a custom build of [Fontello](http://fontello.com/) included. It provided those prev/next links and a couple of custom fonts for tips and warnings.
* There's custom Markdown syntax in form of `T>` and `W>`. That's more semantic than using `>` everywhere.
* Each Markdown header receives an anchor automatically so you can link to a specific part of a page.
* Code highlighting goes through [prism](http://prismjs.com/). There's a bit of custom code but I tried to keep it to minimum. Their Node build is a little problematic and the library works better on frontend. Here I process highlighting through Node so that the build result is completely static apart from third party dependencies (Disqus, Gitter).

### CODE REVIEW

**Comment by mxstbr** [comment]
Love it! Well structured, easy to understand codebase. :+1:

Is there a plan re: styling? Would love to help @gregvenech there!

**Comment by bebraw** [comment]
I think he'll push component styles closer to components (now decoupled). But that's likely a separate PR from him.

**Comment by gregvenech** [comment]
Hey @mxstbr thanks for the offer! My plan right now is to get the antwar branch working on my machine (see #27). Then I want update the splash page code/design on develop and port it over to antwar setup. For now, I'll probably keep the same styling setup as is currently on the antwar branch. Because the component stylesheets are still separated, re-coupling them with their components shouldn't be a big deal.

File: styles/_normalize.scss [hunk 1]
```diff
@@ -0,0 +1,393 @@
+/*! normalize.css v2.1.0 | MIT License | git.io/normalize */
```
Comment (sokra) [inline styles/_normalize.scss]: can we get this from npm instead of including a copy

File: antwar.config.js [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+                require.resolve('globals/globals.json'),
+                require.resolve('regexpu-core/data/iu-mappings.json')
+              ]
+            }
+          ]
+        }
+      };
+    },
+    build: function(plugins) {
+      var ExtractTextPlugin = plugins.ExtractTextPlugin;
```
Comment (sokra) [inline antwar.config.js]: This is weird. Can't we just add extract-text-plugin as dependency?

File: package.json [hunk 3]
```diff
@@ -21,31 +21,40 @@
     "url": "https://github.com/webpack/webpack.io/issues"
   },
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
-    "build": "webpack -p",
-    "dev-server": "webpack-dev-server"
+    "start": "antwar -d",
+    "build": "antwar -b && npm run sitemap",
+    "deploy": "antwar -D",
+    "sitemap": "cd build && sitemap-static --prefix=http://webpack.io.github.io/ > sitemap.xml"
```
Comment (sokra) [inline package.json]: I think the current url is `http://webpack.github.io/webpack.io/`

File: content/10-minute-introduction.md [hunk 4]
```diff
@@ -1,4 +1,5 @@
-# 10 Minute Introduction to Webpack
+---
+title: 10 Minute Introduction to Webpackl
```
Comment (TheLarkInn) [inline content/10-minute-introduction.md]: Typo /wepackl/webpack

File: antwar.config.js (hunk 2 above)
Comment (bebraw) [inline antwar.config.js]: Yeah. I'll push the webpack specific configuration to a file of its own. Now it's getting bundled for legacy reasons but splitting it up will allow me to remove extract-text-plugin from antwar core and keep this cleaner.

Basically I'm going to maintain only minimal configuration at antwar in its next version and push the configuration problem to a package of its own I can use across projects.

**Comment by bebraw** [comment]
I did the changes requested by the comments. Most importantly webpack configuration has been pushed to a file of its own. So if someone wants to give the PR a go, feel free.

**Comment by TheLarkInn** [comment]
LGTM. Super excited about this, it will allow people to develop documentation in a much more rich manner and ease!!!

**Comment by d3viant0ne** [comment]
Digging this @bebraw. Really solid base for all the soon to be really solid documentation :)

File: antwar.config.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    title: 'Webpack',
+    path: function() {
+      return contentCb();
+    },
+    processPage: {
+      url: function(o) {
+        return o.sectionName + '/' + o.fileName.split('.')[0]
+      },
+      content: function(o) {
+        var content = o.file.__content.split('\n').slice(1).join('\n')
```
Comment (sokra) [inline antwar.config.js]: Not clear what this do... Could you add an comment?

Comment (sokra) [inline antwar.config.js]: What is in the first line?

File: layouts/SectionIndex.jsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+export default React.createClass({
+  displayName: 'SectionIndex',
+  render() {
+    const section = this.props.section
+
+    return (
+      <div className="section-index__wrapper">
+        <h1>{section.title}</h1>
+
+        <ul className="section-list">{section.pages().reverse().map((page, i) => {
```
Comment (sokra) [inline layouts/SectionIndex.jsx]: Why `reverse`?

File: antwar.config.js (hunk 5 above)
Comment (bebraw) [inline antwar.config.js]: You mean `root(contentCb)`? The system relies on `require.context` for figuring out what content to load so I need to pass the context there somehow.

I simplified it further and added comments to clear it up.

File: layouts/SectionIndex.jsx (hunk 6 above)
Comment (bebraw) [inline layouts/SectionIndex.jsx]: Fixed. I changed the underlying plugin logic a notch to get rid of that.

File: antwar.config.js (hunk 5 above)
Comment (sokra) [inline antwar.config.js]: I mean `var content = o.file.__content.split('\n').slice(1).join('\n')`. Why is the first line removed?

Comment (bebraw) [inline antwar.config.js]: Ah, that was legacy code. I have headers in my book I need to remove but obviously that's not the case here. Fixed. 👍

**Pull request merged**